### DISCUSSION TITLE
File tnsnames.ora not found

### DISCUSSION THREAD

**Discussion by guilpt** [beeware/briefcase#1105] [Q&A] [answered]
Hello, i am building an application that connects to a oracle database, to do that, i need to provide two files: tnsnames.ora and ewallet.pem.
I tested the connection outside the briefcase enviroment and it worked fine, i let these two files in the same directory of the script file with the command: 
```
connection=oracledb.connect(user='...',
                                    password='...',
                                    dsn='...',
                                    config_dir='',
                                    wallet_location='',
                                    wallet_password='...')
```
the config_dir parameter points the directory where the tnsnames.ora is and the wallet_location points the diretory where the ewallet.pem is.
The thing is, when i run my application with briefcase dev, i got the this error:  'cannot connect to database. File tnsnames.ora not found in '
The library oracledb is already installed and listed in 'requires' parameter at pyproject.toml.
The folder containing both files is listed in 'sources' parameter at pyprojetct.toml
Thanks for your help.

**Comment by freakboy3742** [comment, marked as answer]
The problem is almost certainly that path name you're providing to your database files. 

If you're specifying the filenames as relative paths (and specifying "tnsnames.ora" by itself is a relative path), you are making an implicit assumption about the working directory when the code is running. The working directory of a running Briefcase app isn't guaranteed - it's highly platform dependent. All paths to data files need to be specified as *absolute* paths.

If you're using Toga, the toga App object provides a `paths` attribute that can be used to find common useful locations on the filesystem (e.g., `self.paths.app` is a Path object describing the absolute path to the location of the app module). If you're not using Toga, you'll need to use other approaches, such as the `__file__` attribute of the code module that is performing the operation.

**Reply by guilpt** [reply]
Thanks for your reply :)
I placed the files in the Path(toga.__file__).parent and put this path on the parameters of connect method and worked.
